### PR TITLE
Merge project specific update and fixes

### PR DESCRIPTION
**Pull request by 0Nel**

### CODE REVIEW

**Comment by 0Nel** [comment]
Added updates from project specific repo. removed all project specific content. We definitely need to squash the commits and delete the branch to avoid project specific commits to clutter the git history.

Main changes are:

* debug exec, start, delete and tool scripts for using CD as default exec mode in settings
* for release image build copy startscripts recursively
* reorganize cd build script

File: settings.bash [hunk 1]
```diff
@@ -18,9 +18,10 @@ export DOCKER_REGISTRY_AUTOPULL=false
 
 ### The default release mode to use if no mode paramater is given to ./exec.bash or ./stop.bash
 ### The checked in version should reflect the image status and be the highest availale image (base - devel - release)
-export DEFAULT_EXECMODE="base" # Use this only for setting up the initial devel image (modify setup_workspace.bash)
-# export DEFAULT_EXECMODE="devel" # This is used while deveoping code and preparing a relase
-# export DEFAULT_EXECMODE="release" # use the release as default
+#export DEFAULT_EXECMODE="base" # Use this only for setting up the initial devel image (modify setup_workspace.bash)
+#export DEFAULT_EXECMODE="devel" # This is used while deveoping code and preparing a relase
+#export DEFAULT_EXECMODE="release" # use the release as default
+export DEFAULT_EXECMODE="CD" # use the continuous deployment image as default
```
Comment (planthaber) [inline settings.bash:24]: As this repo is the starting point for development, the execmode should be "base" here

File: startscripts/hello_world [hunk 2]
```diff
@@ -1,5 +0,0 @@
-#!/bin/bash
```
Comment (planthaber) [inline startscripts/hello_world:1]: I think this file should stay for documentation and quick testeing this functioanlity (includes autocomplete)

File: stop.bash [hunk 3]
```diff
@@ -19,10 +19,13 @@ if [ "$1" = "release" ]; then
     EXECMODE="release"
     shift
 fi
-if [ "$1" = "CD" ]; then
+if [ "$1" = "release" ] || [ "$1" = "CD" ]; then
     EXECMODE="release"
     shift
 fi
+if [ "$EXECMODE" == "CD" ]; then
```
Comment (planthaber) [inline stop.bash:26]: == won't work in bash, also it is a duplicate of the test in line 22

File: stop.bash [hunk 4]
```diff
@@ -19,10 +19,13 @@ if [ "$1" = "release" ]; then
...
+    EXECMODE="release"
```
Comment (planthaber) [inline stop.bash:27]: Or the mode here is wrong

File: tools/delete_image.bash [hunk 5]
```diff
@@ -18,6 +18,10 @@ if [ "$1" = "release" ]; then
     EXECMODE="release"
     shift
 fi
+if [ "$1" = "CD" ]; then
+    EXECMODE="CD"
```
Comment (planthaber) [inline tools/delete_image.bash:22]: Id the execmode CD or release?

Comment (planthaber) [reply tools/delete_image.bash:22]: I'm unsure how i implemented it, but as far as i remember the CD execmode was "release" with an override of the image

File: settings.bash (hunk 1 above)
Comment (0Nel) [reply settings.bash:24]: True. Thanks for spotting this. I did set it to CD for the project specific use case..

File: tools/delete_image.bash (hunk 5 above)
Comment (0Nel) [reply tools/delete_image.bash:22]: Yes exactly. The CD execmode overrides the image name and then uses the release exec mode. Problem up until now was that this only happens, if CD is passed explicitely to the script and not if it is the default exec mode. Hence the slight duplication

File: stop.bash (hunk 3 above)
Comment (0Nel) [reply stop.bash:26]: == does work in bash, but for consistency we should stick to either one and therefore I ll change it to a single =.

It is not a duplicate of line 22 as it evaluates, if CD is set as default mode in the settings.bash. The difference in content to the if clause in line 22 is just the shift command, which we could agree on removing for this script as it is not needed, but then I'd remove it for all exec modes.

File: tools/delete_image.bash (hunk 5 above)
Comment (0Nel) [reply tools/delete_image.bash:22]: but you are right. for container deletion the execmode should be set to release for the CD image

File: image_setup/02_devel_image/Dockerfile [hunk 6]
```diff
@@ -7,7 +7,8 @@ ENV DEBIAN_FRONTEND=noninteractive
 
 # generic packages to install non-workspace dependencies ( those are added later )
 RUN apt-get update && apt-get install -y \
-    bash-completion 
+    bash-completion \
+    git
```
Comment (planthaber) [inline image_setup/02_devel_image/Dockerfile:11]: I wouldn't install git by default

**Pull request merged**